homeassistant/components/tuya/switch.py

```diff
@@ -558,6 +558,52 @@
             entity_category=EntityCategory.CONFIG,
         ),
     ),
+    # Dehumidifier
+    # https://developer.tuya.com/en/docs/iot/f?id=K9gf45ts0b8lz
+    "cs": (
+        SwitchEntityDescription(
+            key=DPCode.CHILD_LOCK,
+            name="Child Lock",
+            icon="mdi:account-lock",
+            entity_category=EntityCategory.CONFIG,
+        ),
+        SwitchEntityDescription(
+            key=DPCode.SLEEP,
+            name="Sleep",
+            icon="mdi:power-sleep",
+            entity_category=EntityCategory.CONFIG,
+        ),
+        SwitchEntityDescription(
+            key=DPCode.UV,
+            name="UV Sterilization",
+            icon="mdi:minus-circle-outline",
+            entity_category=EntityCategory.CONFIG,
+        ),
+        SwitchEntityDescription(
+            key=DPCode.HEAT,
+            name="Heating",
+            icon="mdi:hvac",
+            entity_category=EntityCategory.CONFIG,
+        ),
```

Comment:
So, hvac/heating, a fan, swing... isn't this just a thermostat and thus needs the climate platform?

Reply:
I don't see how to use the `DPCode.HEAT` switch on the climate platform... what we would have would be a climate entity with humidity, speed fan and swing control, but the mode would not be compatible (the heat switch doesn't either).

Seems like keeping the humidifier platform and these controls separate is better for the user experience, doesn't it?

Reply:
I dunno, when being able to turn on/off a heating element and set a temperature, the climate platform is what it is for.

The climate platform is not limited to thermostats, but works with things like ovens, refrigerators, or anything else that does hvac :)

Reply:
I prepared some changes in `climate.py` here...
Only the mode would be strange... in the humidifier entity there is the selection of the mode which is the dehumidifier mode in this case.

In weather, the mode would be `Heat` or `Off`... and we would have a select entity with the dehumidifying mode (according to the `dehumidify`, `drying`, `continuous` documentation).

I don't have a virtual device from the `cs` category to test...

Reply:
Thinking… on the climate platform it will be possible to turn on the dehumidifier only with the heat switch on… I'll think if there's an output for that.